app/models/membership_application.rb

```diff
@@ -171,6 +174,11 @@ def se_mailing_csv_str
   end
 
 
+  def self.get_next_membership_number
+    connection.execute("SELECT nextval('membership_number_seq')").getvalue(0,0).to_s
```

Comment:
why do you do `.getvalue(0,0)` ?  and then '.to_s' ?

I just am ignorant about why this is needed.  Are you're setting a default value if nothing is returned?

Reply:
The `connection.execute` method returns a `PG::Result` object. The `#getvalue(tup_num, field_num)` method returns the next number in the sequence which is then converted to the string needed for the `membership_number` field

The sequence always returns a number, so there is no default value provided.

Reply:
This is really good.

Just curious - did you think about doing this by just adding a table ("membership_numbers") to the DB?  Rails, of course, will auto-add an ID and a sequence for that ID.  (not recommending you do that or even think about it .... just curious about this for my own awareness).